xml-parser/src/main/java/org/uniprot/core/xml/uniprot/citation/JournalArticleConverter.java

```diff
@@ -37,9 +39,15 @@ public CitationType toXml(JournalArticle uniObj) {
         CitationConverterHelper.updateToXmlCitatation(xmlUniprotFactory, xmlCitation, uniObj);
         xmlCitation.setType(uniObj.getCitationType().getName());
         xmlCitation.setName(uniObj.getJournal().getName());
-        xmlCitation.setFirst(pageConverter.toXml(uniObj.getFirstPage()));
-        xmlCitation.setLast(pageConverter.toXml(uniObj.getLastPage()));
-        xmlCitation.setVolume(uniObj.getVolume());
+        if (!Strings.isNullOrEmpty(uniObj.getFirstPage()) && !uniObj.getFirstPage().equals("0")) {
+            xmlCitation.setFirst(pageConverter.toXml(uniObj.getFirstPage()));
+        }
+        if (!Strings.isNullOrEmpty(uniObj.getLastPage()) && !uniObj.getLastPage().equals("0")) {
+            xmlCitation.setLast(pageConverter.toXml(uniObj.getLastPage()));
+        }
+        if (!Strings.isNullOrEmpty(uniObj.getVolume()) && !uniObj.getVolume().equals("0")) {
+            xmlCitation.setVolume(uniObj.getVolume());
+        }
 
```

Comment:
If we could take out code `!Strings.isNullOrEmpty(uniObj.getFirstPage()) && !uniObj.getFirstPage().equals("0")` to a method with meaningful name that would be better. e.g. notBlankNotZero